docs/resources/Channel.md

```diff
@@ -501,20 +505,37 @@ Metadata about the interaction, including the source of the interaction and rele
 
 ### Message Reference Object
 
-###### Message Reference Structure
+#### Message Reference Structure
 
 | Field               | Type      | Description                                                                                                                             |
 |---------------------|-----------|-----------------------------------------------------------------------------------------------------------------------------------------|
+| type? \*            | integer   | [type of reference](#DOCS_RESOURCES_CHANNEL/message-reference-types).                                                                   |
 | message_id?         | snowflake | id of the originating message                                                                                                           |
-| channel_id? \*      | snowflake | id of the originating message's channel                                                                                                 |
+| channel_id? \*\*    | snowflake | id of the originating message's channel                                                                                                 |
 | guild_id?           | snowflake | id of the originating message's guild                                                                                                   |
 | fail_if_not_exists? | boolean   | when sending, whether to error if the referenced message doesn't exist instead of sending as a normal (non-reply) message, default true |
 
-\* `channel_id` is optional when creating a reply, but will always be present when receiving an event/response that includes this data model.
+\* If `type` is unset, `DEFAULT` can be assumed in order to match the behaviour before message reference had types.
+In future API versions this will become a required field.
 
-#### Message Types
+\*\* `channel_id` is optional when creating a reply, but will always be present when receiving an event/response that includes this data model.
 
-There are multiple message types that have a message_reference object.  Since message references are generic attribution to a previous message, there will be more types of messages which have this information in the future.
+#### Message Reference Types
+
+Determines how associated data is populated.
+| Type    | Value | Coupled Message Field | Description                                              |
+|---------|-------|-----------------------|----------------------------------------------------------|
+| DEFAULT | 0     | `referenced_message`  | A standard reference used by replies.                    |
+| FORWARD | 1     | `message_snapshot`    | Reference used to point to a message at a point in time. |
+
+`FORWARD` can only be used for basic messages; i.e. messages which do not have strong bindings to a non global entity.
```

Comment:
is it accurate that this is only text-based messages? the wording is a bit hard to parse right now so finding out how to suggest something a little cleaner.